prov/sockets/src/sock_cntr.c

```diff
@@ -228,11 +231,13 @@ void sock_cntr_err_inc(struct sock_cntr *cntr)
 
 static int sock_cntr_add(struct fid_cntr *cntr, uint64_t value)
 {
+	uint64_t new_val;
 	struct sock_cntr *_cntr;
 
 	_cntr = container_of(cntr, struct sock_cntr, cntr_fid);
 	pthread_mutex_lock(&_cntr->mut);
-	atomic_set(&_cntr->value, atomic_get(&_cntr->value) + value);
+	new_val = atomic_add(&_cntr->value, value);
+	atomic_set(&_cntr->last_read_val, new_val);
```

Comment:
```last_read_val``` keeps track of the last read value by app. fi_poll() returns success if the current counter value is different than the last read value (as per discussion in #2240)